kraken-build/src/kraken/std/python/buildsystem/uv.py

```diff
@@ -62,136 +63,143 @@ def unsafe_url(self) -> str:
+                entry["default"] = True
+            if index.explicit:
+                entry["explicit"] = True
+            entries.append(entry)
+        return entries
 
     def to_env(self) -> dict[str, str]:
+        """Convert UV configuration for indexes into a environment variables."""
+
         env = {}
-        if self.primary is not None:
-            env["UV_INDEX_URL"] = self.primary.unsafe_url
-        if self.supplemental:
-            env["UV_EXTRA_INDEX_URL"] = os.pathsep.join(idx.unsafe_url for idx in self.supplemental)
+        uv_indexes = []
+        for index in self.indexes:
+            if index.default:
+                # https://docs.astral.sh/uv/configuration/environment/#uv_default_index
+                env["UV_DEFAULT_INDEX"] = index.unsafe_url
+            else:
+                # https://docs.astral.sh/uv/configuration/environment/#uv_index
+                uv_indexes.append((f"{index.name}=" if index.name != "" else "") + index.unsafe_url)
+
+        if len(uv_indexes) != 0:
+            env["UV_INDEX"] = " ".join(uv_indexes)
         return env
 
 
 class UvPyprojectHandler(PyprojectHandler):
     """Implements the PyprojectHandler interface for UV projects."""
 
-    # TODO: Support `uv.toml` configuration file?
-
-    # PyprojectHandler
+    # TODO: Support global `uv.toml` configuration file?
 
     def get_package_indexes(self) -> list[PackageIndex]:
-        """Maps the UV [`index-url`][1] and [`extra-index-url`][2] options to Kraken's concept of package indices.
-        Note that UV does not support the concept of "aliases" for package indices, so instead the package index alias
-        is ignored and generated automatically based on the hostname and URL hash.
+        """Maps the UV [`index`][1] table, [`index-url`][2] and [`extra-index-url`][3] options to Kraken's concept of
+        package indices. Note that UV does not support the concept of "aliases" for package indices, so instead
+        the package index alias is ignored and generated automatically based on the hostname and URL hash.
 
-        [1]: https://docs.astral.sh/uv/reference/settings/#index-url
-        [2]: https://docs.astral.sh/uv/reference/settings/#extra-index-url
+        [1]: https://docs.astral.sh/uv/reference/settings/#index
+        [2]: https://docs.astral.sh/uv/reference/settings/#index-url
+        [3]: https://docs.astral.sh/uv/reference/settings/#extra-index-url
         """
 
-        def gen_alias(url: str) -> str:
-            hostname = urlparse(url).hostname
-            assert hostname is not None, "expected hostname in package index URL"
-            return f"hostname-{md5(url.encode()).hexdigest()[:5]}"
-
         indexes: list[PackageIndex] = []
-        config: dict[str, Any] = self.raw.get("tool", {}).get("uv", {})
+        for index in self.raw.get("tool", {}).get("uv", {}).get("index", []):
+            indexes.append(
+                PackageIndex(
+                    alias=index.get("name", ""),
```

Comment:
we might want to us a special string instead of `""` to make debugging easier

Reply:
Or `None` :p

Reply:
While other package managers like `pdm`, `poetry` and `slap` require that their indexes are named, `uv` doesn't. Hence, `alias` is marked as a required string. I originally wanted to represent the the absence of the index name with `None`, but this had cascading effects on all the call sites for mypy. Indeed, other build systems or environment managers would have to ensure `alias` is not known every time they consume that value. Although maybe sub-optimal, I decided to limit the depth of the changes and use an empty.